### PR TITLE
Expose JVM exceptions

### LINKED ISSUE #472: [FEATURE REQUEST]: Expose JVM exceptions
Currently an exception is only logged and then translated per the below.
```
[2020-04-03T00:31:33.1953860Z] [0ecc54d4f7e9] [Error] [JvmBridge] JVM method execution failed: Nonstatic method sql failed for class 435 when called with 1 arguments ([Index=1, Type=String, Value=select
 NOTACOLUMN from tmp], )
[2020-04-03T00:31:33.1955104Z] [0ecc54d4f7e9] [Error] [JvmBridge] org.apache.spark.sql.AnalysisException: cannot resolve '`NOTACOLUMN`' given input columns: [tmp.Parameter]; line 1 pos 7;
'Project ['NOTACOLUMN]
+- SubqueryAlias `tmp`
   +- Relation[Parameter#20315] parquet<...SNIP...>
```
Is it possible maybe throw an AggregateException with both exceptions or pass an InnerException containing the full original exception?

### PR DESCRIPTION
**Pull request by thrixton**
Wrap spark exceptions
Fixes #472
Ref also #541, #502

### CODE REVIEW

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/JvmBridgeTests.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark.E2ETest/IpcTests/JvmBridgeTests.cs:1]: Can you add the license header ?

File: src/csharp/Microsoft.Spark/JvmException.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark/JvmException.cs:1]: Can you add the license header ?

File: src/csharp/Microsoft.Spark/JvmException.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace Microsoft.Spark
+{
+    /// <summary>
+    /// Contains the message returned from the <see cref="Interop.Ipc.JvmBridge"/> on an error
```
Comment (suhsteve) [inline src/csharp/Microsoft.Spark/JvmException.cs:6]:
```suggestion
    /// Contains the message returned from the <see cref="Interop.Ipc.JvmBridge"/> on an error.
```

File: src/csharp/Microsoft.Spark.E2ETest/IpcTests/JvmBridgeTests.cs (hunk 1 above)
Comment (thrixton) [reply src/csharp/Microsoft.Spark.E2ETest/IpcTests/JvmBridgeTests.cs]: Done

File: src/csharp/Microsoft.Spark/JvmException.cs (hunk 2 above)
Comment (thrixton) [reply src/csharp/Microsoft.Spark/JvmException.cs]: Done

**Pull request merged**